awx/ui_next/src/api/Base.js

```diff
@@ -10,6 +10,16 @@ const defaultHttp = axios.create({
   },
 });
 
+defaultHttp.interceptors.response.use(response => {
+  const timeout = response?.headers['session-timeout'];
+  if (timeout) {
+    const timeoutDate = new Date().getTime() + timeout * 1000;
+    window.localStorage.setItem('session-timeout', String(timeoutDate));
```

Comment:
GUI initiated HTTP activity tends to be bursty, and so we _may_ want to debounce the call to `setItem` so that we only hit the storage on the last response.

It shouldn't be a problem to add this type optimization later by wrapping the function in a debounce utility. That said, I intend to do some preliminary testing on pages with relatively high amounts of chatter (Job Templates, Credentials, etc.) prior to merging.

Reply:
I decided to debounce this function call. While I don't think it's a situation that users will encounter, I noticed some slowdown with our auto-refreshing dev server on firefox when many tabs were opened and this helped.